src/ai/backend/manager/models/gql_models/container_registry_utils.py

```diff
@@ -0,0 +1,251 @@
+            previous_quota, quota_id = quota_info["previous_quota"], quota_info["quota_id"]
+
+            if previous_quota > 0:
+                raise GenericBadRequest(f"Quota limit already exists. (gr: {self.project_id})")
+
+            put_quota_api = yarl.URL(self.registry.url) / "api" / "v2.0" / "quotas" / str(quota_id)
+            payload = {"hard": {"storage": quota}}
+
+            async with sess.put(
+                put_quota_api, json=payload, allow_redirects=False, **rqst_args
+            ) as resp:
+                if resp.status != 200:
+                    log.error(f"Failed to create quota! response: {resp}")
+                    raise InternalServerError(f"Failed to create quota! response: {resp}")
+
+    async def update(self, quota: int) -> None:
+        ssl_verify = self.registry.ssl_verify
+        connector = aiohttp.TCPConnector(ssl=ssl_verify)
+        async with aiohttp.ClientSession(connector=connector) as sess:
+            rqst_args: dict[str, Any] = {}
+            rqst_args["auth"] = aiohttp.BasicAuth(
+                self.registry.username,
+                self.registry.password,
+            )
+
+            quota_info = await self._get_quota_info(sess, rqst_args)
+            previous_quota, quota_id = quota_info["previous_quota"], quota_info["quota_id"]
+
+            if previous_quota == -1:
+                raise ObjectNotFound(object_name="quota entity")
+
+            put_quota_api = yarl.URL(self.registry.url) / "api" / "v2.0" / "quotas" / str(quota_id)
+            payload = {"hard": {"storage": quota}}
+
+            async with sess.put(
+                put_quota_api, json=payload, allow_redirects=False, **rqst_args
+            ) as resp:
+                if resp.status != 200:
+                    log.error(f"Failed to update quota! response: {resp}")
+                    raise InternalServerError(f"Failed to update quota! response: {resp}")
+
+    async def delete(self) -> None:
+        ssl_verify = self.registry.ssl_verify
+        connector = aiohttp.TCPConnector(ssl=ssl_verify)
+        async with aiohttp.ClientSession(connector=connector) as sess:
+            rqst_args: dict[str, Any] = {}
+            rqst_args["auth"] = aiohttp.BasicAuth(
+                self.registry.username,
+                self.registry.password,
+            )
+
+            quota_info = await self._get_quota_info(sess, rqst_args)
+            previous_quota, quota_id = quota_info["previous_quota"], quota_info["quota_id"]
+
+            if previous_quota == -1:
+                raise ObjectNotFound(object_name="quota entity")
+
+            put_quota_api = yarl.URL(self.registry.url) / "api" / "v2.0" / "quotas" / str(quota_id)
+            payload = {"hard": {"storage": -1}}  # setting quota to -1 means delete
+
+            async with sess.put(
+                put_quota_api, json=payload, allow_redirects=False, **rqst_args
+            ) as resp:
```

Comment:
Please check this part as well.

Reply:
This has already been discussed in Teams, but refer to the Harbor API documentation for detailed information.